### PR TITLE
Address issue #489: Deprecate enabling nested parallelism

### PR DESCRIPTION
**Pull request by cndolo**
In this PR the `enableNestedParallelism` function is deprecated.

### CODE REVIEW

**Comment by cndolo** [comment]
@angriman  `__init.py` calls the now deprecated function `enableNestedParallelism()`. I thought that should be removed too, and added that to the PR. What do you think?

File: networkit/_NetworKit.pyx [hunk 1]
```diff
@@ -241,6 +241,8 @@ def getMaxNumberOfThreads():
 
 def enableNestedParallelism():
 	""" Enable nested parallelism for OpenMP"""
+	from warnings import warn
+	warn("Nested parallelism has been deprecated.")
 	_enableNestedParallelism()
```
Comment (angriman) [inline networkit/_NetworKit.pyx]: You can remove this line, we do not allow nested parallelism anymore

File: networkit/__init__.py [hunk 2]
```diff
@@ -161,7 +161,6 @@ def setup():
 	""" This function is run once on module import to configure initial settings """
 	setLogLevel("ERROR")  # set default loglevel for C++ code
 	setPrintLocation(True)
-	enableNestedParallelism()  # enable nested parallelism
```
Comment (angriman) [inline networkit/__init__.py]: This line should not be removed, just deprecate the method. We should remove it in future releases.

File: networkit/_NetworKit.pyx (hunk 1 above)
Comment (cndolo) [reply networkit/_NetworKit.pyx]: I have, and restored `__init.py` to its initial state.

Comment (angriman) [reply networkit/_NetworKit.pyx]: Sorry my bad that change was ok.

Comment (cndolo) [reply networkit/_NetworKit.pyx]: OK, it's back.

**Pull request merged**